iml-api/src/graphql.rs

```diff
@@ -143,6 +144,94 @@ impl QueryRoot {
 
         Ok(xs)
     }
+    #[graphql(arguments(
+        limit(description = "paging limit, defaults to 20"),
+        offset(description = "Offset into items, defaults to 0"),
+        dir(description = "Sort direction, defaults to asc"),
+        args(description = "Snapshot listing arguments")
+    ))]
+    /// Fetch the list of snapshots
+    async fn snapshots(
+        context: &Context,
+        limit: Option<i32>,
+        offset: Option<i32>,
+        dir: Option<SortDir>,
+        args: List,
+    ) -> juniper::FieldResult<Vec<Snapshot>> {
+        let dir = dir.unwrap_or_default();
+
+        if args.detail {
+            let xs = sqlx::query!(
+                r#"
+                    SELECT filesystem_name, snapshot_name, create_time, modify_time, snapshot_fsname, mounted, comment from snapshot s
+                    WHERE filesystem_name = $4 AND $5::text IS NULL OR snapshot_name = $5
+                    ORDER BY
+                        CASE WHEN $3 = 'asc' THEN s.snapshot_name END ASC,
+                        CASE WHEN $3 = 'desc' THEN s.snapshot_name END DESC
+                    OFFSET $1 LIMIT $2"#,
+                offset.unwrap_or(0) as i64,
+                limit.unwrap_or(20) as i64,
+                dir.deref(),
+                args.fsname,
+                args.name,
+            )
+            .fetch_all(&context.client)
+            .await?;
+
+            let snapshots: Vec<_> = xs
+                .into_iter()
+                .map(|x| Snapshot {
+                    snapshot_name: x.snapshot_name,
+                    filesystem_name: x.filesystem_name,
+                    details: vec![Detail {
+                        comment: x.comment,
+                        create_time: x.create_time,
+                        modify_time: x.modify_time,
+                        snapshot_fsname: x.snapshot_fsname,
+                        // FIXME
+                        snapshot_role: None,
+                        status: x.mounted.map(|b| {
```

Comment:
I am ok with getting rid of `Status` and using `Option<bool>`. Don't know what's better.

Reply:
I think we either need to make `Unknown` a variant in the enum and get rid of the `Option`, or get rid of the enum. Depends if there are other statuses besides `Mounted`  and `Unmounted`.

Reply:
If you had the `mount_path` as an option you could check if it is a `some` and if it is that would evaluate to a state of mounted.

Reply:
The other option is that you could implement the `From` trait and then do:
```
status: x.mounted.into()
```

Reply:
Why is there an option currently in 

https://github.com/whamcloud/integrated-manager-for-lustre/blob/ce8f0c996d49a59a7105c826410fc86fc7cfcf8a/iml-wire-types/src/snapshot.rs#L50

Is it possible to have a `None`?